source/common/umf_pools/disjoint_pool_config_parser.cpp

```diff
@@ -118,34 +163,46 @@ DisjointPoolAllConfigs parseDisjointPoolConfig(const std::string &config,
                                                 DisjointPoolMemType memType =
                                                     DisjointPoolMemType::All) {
         bool ParamWasSet;
+        size_t TmpValue = 0;
+        umf_result_t ret = UMF_RESULT_SUCCESS;
         DisjointPoolMemType LM = memType;
         if (memType == DisjointPoolMemType::All) {
             LM = DisjointPoolMemType::Host;
         }
 
-        bool More = ParamParser(Params, AllConfigs.Configs[LM].MaxPoolableSize,
-                                ParamWasSet);
+        bool More = ParamParser(Params, TmpValue, ParamWasSet);
         if (ParamWasSet && memType == DisjointPoolMemType::All) {
             for (auto &Config : AllConfigs.Configs) {
-                Config.MaxPoolableSize = AllConfigs.Configs[LM].MaxPoolableSize;
+                ret = umfDisjointPoolParamsSetMaxPoolableSize(Config, TmpValue);
+                CheckConfigRet(ret);
```

Comment:
This return will only exit from this lambda, not from the entire parseDisjointPoolConfig function.

Reply:
Done